src/decanter/core/core_api/predict_input.py

```
@@ -49,21 +61,36 @@ def __init__(
         self.pred_body = CoreBody.PredictBody.create(
             data_id='tmp_data_id', model_id='tmp_model_id', callback=callback,
             keep_columns=keep_columns, threshold=threshold, version=version)
+        self.select_model = select_model
+        self.select_opt = select_opt
 
     def getPredictParams(self):
         """Using pred_body to create the JSON request body for prediction.
 
         Returns:
             :obj:`dict`
         """
+        if self.select_model == 'best':
+            select_model_id = self.experiment.best_model.id
+        elif self.select_model == 'model_id':
+            if self.select_opt in self.experiment.models:
+                select_model_id = self.select_opt
+            else:
+                logger.error('[%s] Invalid input model ID', self.__class__.__name__, self.select_opt)
```

should we raise a exception over here?